### PR TITLE
Addition of DQN Trainer and Integration with DFDX Library

### PR DESCRIPTION
**Pull request by chriamue**
This pull request introduces a DQN Trainer. It is primarily employed in the eucdist_dqn example and serves as an important step towards leveraging more advanced machine learning strategies. I welcome all discussions, suggestions, and critiques regarding this addition.

Please note that the DQN Trainer implementation necessitates the inclusion of a new dependency, the machine learning library 'DFDX'. This dependency is solely required when the 'DQN' feature is enabled, ensuring that it will not impact users who do not use this specific feature.

The code is partly based on the dfdx rl-dqn example https://github.com/coreylowman/dfdx/blob/main/examples/rl-dqn.rs

Feel free to share your thoughts. Whether it's a suggestion for improvement or a disagreement with the proposed changes.

Looking forward to your reviews. Thank you in advance.

### CODE REVIEW

**Comment by milanboers** [comment]
Hi! Thanks for the PR and sorry for the late response!

Being able to easily swap out elements of the training strategy without having to redefine your problem is just what this package is about, so this seems like a great addition :-) I also like that the dependency is optional.

I wonder if there's ways to integrate it with the existing `AgentTrainer` somehow, so it's not needed to swap out the entire trainer (like a `LearningStrategy` for DQN with a `TerminationStrategy` for the batching), but I realize this can be tricky to implement.
I also see some hardcoded values (batch size, learning rate, gamma) and I wonder if they work for every problem.

But I'll go ahead and merge this. I'm fixing a small compilation issue in the example, and some clippy issues, and then I'll cut a new release!
